packages/docs/docs/guides/docsqa.md

```diff
@@ -0,0 +1,82 @@
+# Grounding Answers with a Source of Truth - DocsQA
+
+Large Language Models (LLMs) are powerful tools for answering open domain questsions, but they're prone to making things up ("hallucination") when their responses are unrestricted. In addition, publicly available LLMs are trained on public data and consequently won't have knowledge of your internal documentation.
+
+One of the common ways to handle both of these issues is to provide the LLM with the content of documents that should be treated as a source of truth. This is called "Document Q&A" or "DocsQA." Simple prompting along with document context avoids hallucination by restricting the domain of responses. At the same time, providing new knowledge from internal sources allows the LLM to give more useful responses.
+
+## Overview
+
+A collection of docs is called a "corpus" (plural "corpora"). Often corpora are backed by a "vector database" that allows for searching over knowledge in a semantically grounded vector space. To use a corpus with an LLM, you'll have a process like the one below. (Don't worry if this doesn't all make sense yet - we'll explain it!)
+
+```mermaid
+---
+title: Preparing a Corpus
+---
+flowchart LR
+  Load -->|bytes| Parse -->|Document| Chunk -->|Text chunks| Embed -->|vectors| vdb[(Vector DB)]
+```
+
+```mermaid
+---
+title: Handling a Query
+---
+flowchart LR
+  que[Query] -->|string| embed2[Embed] -->|vector| vdb2[(Vector DB)] -->|similar chunks| LLM
+```
+
+## Preparing a Corpus
+
+In order for a corpus to find text relevant to a given query, it first needs to be prepared with the text it may return. There are a few steps involved:
+
+### Step 1: Loading (and Parsing)
+
+Step 1 is identifying the documents you care about. In the simplest case, this may be a static list of text files you've already downloaded that you can simply read into memory. In more complex cases, this could involve crawling the web or reading from a database.
+
+Regardless of where your documents reside, you'll need to implement a "Loader" to bring them into your program. The "Loader" interface provided by ai.jsx allows for loading documents from any source with appropriate parallelization and pagination.
+
+For non-text documents (PDFs, Word docs, podcasts, etc.), you'll need to convert the raw document bytes into a text representation. In this case, you'll use ai.jsx's "RawLoader" along with a "Parser" where the former is responsible for finding the document and getting it into memory and the latter is responsible for turning it into text.
+
+Note: When using a Fixie corpus, Fixie can handle parsing for most common documents types for you.
```

Comment:
You can use https://docusaurus.io/docs/markdown-features/admonitions to have a richer note block.

Reply:
Thanks, that's just what I was looking for. :)